aws_lambda_powertools/shared/user_agent.py

```diff
@@ -108,6 +108,31 @@ def register_feature_to_session(session, feature):
         logger.debug(f"session passed in doesn't have a event system:{e}")
 
 
+# Add feature user-agent to given sdk botocore.session.Session
+def register_feature_to_botocore_session(botocore_session, feature):
```

Comment:
Where do we use this function?

IIRC we already had one function to add user-agent to the session.

Btw, we also need to ignore this file from test coverage (unless we're confident we can unit test them properly)

Reply:
Hi Heitor, Sorry I added the reason in issue but forgot to add it here. 

- Where do we use this function?
  - For supporting Seshu's data-masking pr. Which uses AWS KMS as one of it's provider [here](https://github.com/aws-powertools/powertools-lambda-python/pull/2197/files#diff-d258c64782d47c37745e1f7cbba3914710a0ec9832ae1a3e9b3e377b5c896d55R40). 
- IIRC we already had one function to add user-agent to the session.
  - The special thing about kms is it uses a botocore session instead of boto3 session to [init it's client](https://github.com/aws/aws-encryption-sdk-python/blob/560e7143ac7caf98e190b17ce2af97b7eea6be16/src/aws_encryption_sdk/key_providers/kms.py#L671C60-L671C60). So we need a new function to patch the new botocore session.

Reply:
Ah that makes sense now.

Could you add that info in the docstring and update the PR body with the reasoning?

We can merge afterwards otherwise we'll miss this info in the PR

Reply:
Added to pr body and modified the docstring.

Reply:
Thank you so much @roger-zhangg - merging now